modules/ROOT/pages/design-common-errors-08.adoc

```diff
@@ -0,0 +1,360 @@
+
+
+----
+#%RAML 0.8
+title: ExampleRAML
+
+/upload:
+  post:
+    description: |
+      Upload a photo
+    body:
+      multipart/form-data:
+        formParameters:
+          title:
+            description: The title of the photo.
+          is_public:
+            type: boolean
+            example: 1
+----
+
+== Common Error 7
+// APIMF-1023
+*_Using absolute paths to included files_*
+
+Paths to included files must be relative. The following two snippets together give an example of the error. The `traits` node in the specification `api.raml` includes the file `traits.raml`, and correctly includes it with a relative path. However, the file `traits.raml` includes an example that is located in the file `common_400.example`. However, the `!include` statement uses an absolute path. The error is flagged in `api.raml` at the `traits` node.
+
+The `!include` statement in `traits.raml` should use either `./common/common_400.example` or `common/common_400.example`, rather than the absolute path.
+
+./api.raml
+
+----
+#%RAML 0.8
+title: ExampleRAML
+traits: !include ./common/traits.raml
+
+/booking/list:
+    is: [common_errors]
+    post:
+        body:
+            application/json:
+                example: {}
+----
+
+./common/traits.raml
+
+----
+- common_errors:
+    responses:
+      400:
+        body:
+          application/json:
+            example: !include /common/common_400.example
+----
+
+
+== Common Error 8
+// APIMF-1069
+*_Using invalid JSON in examples of JSON schemas_*
+
+Examples of JSON schemas must be valid, unlike the example in the following snippet:
+
+
+----
+#%RAML 0.8
+title: ExampleRAML
+...
+/api:
+  get:
+    responses:
+      200:
+        body:
+          application/json:
+            schema:
+              {
+                "type": "object",
+                "required": true,
+                "$schema": "http://json-schema.org/draft-03/schema",
+                "properties": {
+                  "a": {
+                    "type": "boolean",
+                    "required": true
+                  }
+                }
+              }
+            example:
+              {
+                "a: {
+                  "a": ""
+                }
+----
+
+
+== Common Error 9
+// APIMF-1083
+*_Not providing a value for the `title` node_*
+
+The `title` node cannot lack a value, as it does here:
+
+
+----
+#%RAML 0.8
+title:
+----
+
+== Common Error 10
+// APIMF-1088
+*_Not using in an example of a query parameter the data type that is in the definition of the query parameter_*
+
+For example, if the type for a query parameter is string, then the example for the query parameter must also be string. Here, the example is an integer, but the query parameter is defined as a string.
+
+
+----
+#%RAML 0.8
+title: ExampleRAML
+/books:
+  get:
+    queryParameters:
+      publicationYear:
+        type: string
+        example: 2016
+----
```

Comment:
In fact, is for all types (each parameter contains a type definition) not only for queryParameters.
Also,  it's the same case that the one in "Common Error 1". The only difference is that here you use an integer example and in the first one you use a boolean example.

Reply:
Alright, then I'll consolidate the two errors into a more general discussion about examples needing to use the same type as the things that they are examples of.

Reply:
Actually, isn't error #1 a different type of problem? In that error, no matter whether the enum uses scalar or string values, the example must be a string. That seems different from the general claim that examples must use the same type as the things that they are examples of.